src/items.rs

```diff
@@ -146,11 +154,21 @@ impl Rewrite for ast::Local {
                     std::cmp::min(shape.width, context.config.single_line_let_else_max_width());
 
                 // If available_space hits zero we know for sure this will be a multi-lined block
-                let available_space = max_width.saturating_sub(result.len());
+                let assign_len = if context.config.version() == Version::Two {
+                    result.len() - let_kw_offset
+                } else {
+                    result.len()
+                };
+                let available_space = max_width.saturating_sub(assign_len);
 
+                let assign_str = if context.config.version() == Version::Two {
+                    &result[let_kw_offset..]
+                } else {
+                    result.as_str()
+                };
                 let allow_single_line = !force_newline_else
                     && available_space > 0
-                    && allow_single_line_let_else_block(&result, block);
+                    && allow_single_line_let_else_block(assign_str, block);
```

Comment:
Thanks for looking into this change. I think it's simpler to rewrite this as follows:

```rust
    let assign_str_with_else_kw = if context.config.version() == Version::Two {
        &result[let_kw_offset..]
    } else {
        result.as_str()
    };

    let available_space = max_width.saturating_sub(assign_str_with_else_kw.len());

    let allow_single_line = !force_newline_else
        && available_space > 0
        && allow_single_line_let_else_block(assign_str_with_else_kw, block);
```

Reply:
That's simpler. Thank you for the suggestion :+1: